lib/post_creator.rb

```diff
@@ -397,6 +399,27 @@ def unarchive_message
     end
   end
 
+  def remove_useless_quotes
+    num_quotes = @post.raw.scan(/\[quote[^\]]*\]/).size
+    return if num_quotes != 1
+
+    prev = @post.topic.posts.order('post_number desc').where('post_number < ? AND not hidden AND posts.deleted_at IS NULL', @post.post_number).first
```

Comment:
Should we check if `post_number` is 1?

Reply:
maybe just `pluck` raw here is is more efficient `.limit(1).pluck(:raw).first` you don't really need any of the other post data